peer/brontide.go

```diff
@@ -1836,8 +1838,6 @@ out:
 		}
 	}
 
-	p.wg.Done()
```

Comment:
Slight behaviour change here:
- Previously: `wg.Done` then disconnect
- Now: disconnect then `wg.Done`

Seems like this new ordering is the right way, but perhaps just check that nothing strange happens with the change?

Reply:
Ah, good catch! I looked at the initial commit message that added the `p.wg.Done()` there and it explicitly mentioned a deadlock. So I reverted this change and replaced it with that comment instead.